### DISCUSSION TITLE
Glitches in some games (video)

### DISCUSSION THREAD

**Discussion by c0n5um3r** [hyprwm/Hyprland#6771] [Q&A]
Hi! 
I'm getting glitches in Cyberpunk 2077 and in GTA V (but a lot less severe I must say). 
In Cyberpunk 2077 it looks like this: https://youtu.be/E_SFYiW6RCM

Also I tested those games on Plasma 6.1 (on Wayland) with the same settings and on Plasma they work flawless. 
Is something I'm missing in Hyprland cofigs? I tried to use Gamescope and various Steam launch options, but without any luck unfortunately. Also I tried few versions of Proton (and ProtonGE) - nothing changes. I have full AMD setup (CPU and GPU) and I'm on Arch. I use laptop without any additional monitors connected.

Can someone please help me figure out how to fix those glitches?

Other games like Red Dead Redemption 2, Ready or Not, Mount & Blade Bannerlord works great on Hyprland. That's the games I tested at least.

**Comment by ItsLiyua** [comment]
I'm experiencing a similar issue. The game isn't stuttering until I turn too far. Then it starts. When I turn back the stuttering stops briefly. When I keep turning it starts again. My theory is that this happens, when the mouse cursor, even tho it is invisie goes over the edge of the screen and gets moved back by hyprland. It didn't happen when the game was in a window, that had a lot of space left and right of it (probably because of no screen edge to hit there so the reset worked fine). Playing windowed with a 10px border gap around the window lead to the same stuttering but only when turning faster, which supports my theory since the mouse would have to move more than those 10px in one mouse-handling-intervall (idk how to word it any better) to interfere with the screen edge. Can you confirm this behavior @c0n5um3r ?

**Reply by c0n5um3r** [reply]
Sorry for late answer, I had lots of work to do.

> My theory is that this happens, when the mouse cursor, even tho it is invisie goes over the edge of the screen and gets moved back by hyprland.

Yeah, I had same thoughts.

> Can you confirm this behavior @c0n5um3r ?

Unfortunately, no. 

Did you tried this solution from @xadips ?

>Had the same issue before but downgrading mesa from 24.1 down to 24.0.8 helped same as downgrading xwayland to 23.2.7 or older.

**Comment by xadips** [comment]
Had the same issue before but downgrading mesa from 24.1 down to 24.0.8 helped same as downgrading xwayland to 23.2.7 or older.

**Comment by c0n5um3r** [comment]
Quick update: setting Vsync to 55 in Cyberpunk 2077 fixes the issue.
My monitor refresh rate in 165 Hz btw (and it is mentioned in my Hyprland config file).

**Comment by ItsLiyua** [comment]
This got fixed in v42.0 I think. Not quite sure which version but it's fixed now.
